### PR TITLE
Update version and changelog for 0.8.0 release

### PR DESCRIPTION
**Pull request by dazuma**

### CODE REVIEW

**Comment by TheRoyalTnetennba** [comment]
Do you want to throw in some of the updates dependabot is pushing for? Otherwise, LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.8.0 (2019/01/02)
+
+* Support connection options :default_connection and :connection_builder when creating credentials that need to refresh OAuth tokens. This lets clients provide connection objects with custom settings, such as proxies, needed for the client environment.
+
```
Comment (blowmage) [inline CHANGELOG.md]: Do you want to mention the removal of the project_id warning in the release notes?

Comment (dazuma) [reply CHANGELOG.md]: Done.

**Comment by dazuma** [comment]
Per offline discussion, we'll release as 0.8.0 since we need to change downstream dependencies anyway to support 0.7.x.

**Pull request merged**